### ISSUE TITLE
Migrate Supporters service from Bucket to Walrus

### ISSUE THREAD

**Issue by DarrenN** [thegroundwork/groundwork.js#43]
## Background

For October release we will be moving all new orgs to Walrus for data collection, replacing Bucket. Existing orgs will still use Bucket until they are migrated.
## Spec
1. Determine how to branch users between Walrus and Bucket (API version?)
2. Setup calls to Walrus for Supporters service. Will be purpose-built at first until a general Walrus service is written for gw.js

**Comment by nalesandro** [comment]
From a conversation today and worth considering:  Instead of forcing devs to `GET schemaId` and populate their supporters service call with it (a lot to do just to post an email address), should gw.js do something like this:
- User can optionally specify `schemaId` in `supporters.create`
- If they do, then the walrus API call uses that schema when creating a walrus record (duh)
- If not, gw.js makes a call to walrus to `GET schemaId` and then uses the newest schema in the subsequent call to create a walrus record

**Comment by nalesandro** [comment]
Relevant:  https://gist.github.com/nalesandro/7e75364abf4515cf1cc1053cbb8d073f

**Comment by DarrenN** [comment]
Noted, but aren't we kind of punting on this for a bit due to the work @strife25 is doing to point old Bucket to Walrus?

**Comment by strife25** [comment]
> Noted, but aren't we kind of punting on this for a bit due to the work @strife25 is doing to point old Bucket to Walrus?

This is separate - what Nick is noting here is trying to make the step of setting up walrus in one's code easier to do by having gw.js dynamically retrieving the Schema ID for creating a record so that the programmer does not have to do so.

Looking at these steps typed out, I would recommend not doing this as it's effectively creating the idea of a "default" schema. This is nice for getting someone up and running, but if the customer/coder ever changes the schema to a new version that is incompatible, then the code using the above algorithm will break.

Even though it isn't a nicer experience for brand new users, I think it's better to help protect users from accidentally breaking their own websites (and blaming us for the problem) - which is why the API requires the schema ID to be declared. Yes it sucks for band new users, but I'd rather add an extra step to the setup process than create a situation where their old websites get broken. 

I think the better thing to do would be to make it easy to get the schema IDs. I know that's easier said than done with no walrus UI in place atm, but why not just include a `curl` statement in the docs as a step they need to take when using gw.js? For example:

```
curl -X "GET" "https://api.dev.thegroundwork.com/collections/supporters/schemas/" \
     -H "Content-Type: application/json" \
     -H "x-gw-initiative: <YOUR ORG ID>." \
     -u <YOUR API KEY>
```

Thoughts?

**Comment by DarrenN** [comment]
Its going to take me a while to respond to this with the correct amount of thought as I haven't had any time to dig into Walrus to date.

**Comment by nalesandro** [comment]
(for later when D$ has a chance to dig in)

You are reading it more or less correctly @strife25 - I'm advocating an approach where gw.js users can optionally use a default schema, but for the supporters collection **only**.  I get not supporting it in the core service, but one of gw.js's _raisons d'être_ is to simplify the developer's life by abstracting away some of the complexity of the API and this is a good example of where we can/should, IMO. 

The case you're concerned with is indeed possible, but the VAST majority of customers using gw.js to post to the supporters collection will never need to change the canned schema, and in the tiny number of cases where they do, a breaking change is far less likely than a non-breaking one (ie cust will want to add a field vs remove one).  

So for me it comes to a tradeoff between a probable "WTF" for every new customer who just wants to post a dang email address to a supporter service vs an improbable "shit we broke things" which can be mitigated by docs and support

**Comment by DarrenN** [comment]
Picking this up

**Comment by nalesandro** [comment]
🙌

**Comment by DarrenN** [comment]
Closed in #52
